### DISCUSSION TITLE
A simple HTML gallery

### DISCUSSION THREAD

**Discussion by rnkn** [beetbox/beets#4063] [Show and tell]
Given I use console music players, for a while I've missed perusing my music collection by artwork. No longer!

This little script sends an HTML gallery to stdout. 

```sh
#!/bin/sh

cover_width="${COVER_WIDTH:-220}"
album_sort="${ALBUM_SORT:-albumartist+}"
tmpfile=$(mktemp); trap "rm -f $tmpfile" EXIT

beet ls -a "$album_sort" -f '$albumartist|$album|$artpath' |
	while read line; do
		artist=$(echo "$line" | cut -d\| -f1)
		album=$(echo "$line" | cut -d\| -f2)
		artpath=$(echo "$line" | cut -d\| -f3)
		[ -f "$artpath" ] && cat <<EOF
<div class="cover" style="background-image: url(&quot;${artpath#$PWD/}&quot;);">
<div class="details">
	<p>$artist<br/>$album</p>
</div>
</div>
EOF
		>> "$tmpfile"
	done

cat <<EOF
<!DOCTYPE html>
<html>
<head>
<style type="text/css">
	div.cover {
		float: left;
		position: relative;
		width: ${cover_width}px;
		height: ${cover_width}px;
		margin: 5px;
		background-position: center;
		background-size: cover;
	}
	div.details {
		visibility: hidden;
		width: 100%;
		height: 100%;
		background-color: rgba(224,224,224,0.75);
		position: absolute;
	}
	div.cover:hover div.details {
		visibility: visible;
	}
	div.details p {
		font-family: sans-serif;
		text-align: center;
		position: absolute;
		bottom: 0;
		width: 100%;
	}
</style>
</head>
<body>
$(cat "$tmpfile")
</body>
</html>
EOF

rm -f "$tmpfile"
```

Usage (in library directory):
```
$ ALBUM_SORT=year- beet_gallery > index.html
$ python -m http.server -d .
```

**Comment by matlads** [comment]
Very Neat. I like it!

I just added two lines, to support utf-8 characters.

`<meta charset="utf-8">`

**Reply by rnkn** [reply]
Ah right, setting the browser's default encoding to UTF-8 is something I've done for so long I didn't even consider this!

**Comment by RollingStar** [comment]
No screenshot?

**Reply by rnkn** [reply]
It looks like a bunch of album covers.

**Comment by valrus** [comment]
I just published a beets plugin (based on the web plugin) that does this:

https://pypi.org/project/beetsplug-covergrid/

It's real basic right now: random order, no interactivity. But I might expand it in the future...

**Reply by RollingStar** [reply]
Screenshot would be appreciated.

**Reply by valrus** [reply]
The [homepage](https://source.tube/valrus/beetsplug-covergrid) has one but there is also one on the PyPI project page now. But here it is for your convenience:
<img width="1659" height="1454" alt="image" src="https://github.com/user-attachments/assets/d5e0b53d-c259-4505-b157-81deb229abde" />
